### DISCUSSION TITLE
Sleek v2 users feedbacks thread

### DISCUSSION THREAD

**Discussion by lrq3000** [ransome1/sleek#562] [General]
This thread aims to collect feedback from users of the v2 branch, since the first stable release rolled out yesterday. Please mention if you were a user of v1 too or if you started with v2.

/EDIT: This is not a support thread, bugs should be reported as an Issue (please search for an old one if it exists before creating a new one). Feedbacks here are meant to pertain to the design decisions for the v2 branch: what you like, what you miss from v1, what you would like as new features optionally. Bugs can still be mentioned on the side if they did not exist in v1 but you should not expect support here. Please remember the project is mostly developed by a single volunteer, so it is unrealistic to expect free support too.

**Comment by lrq3000** [comment]
I'll start with my own feedback: I discovered the todo.txt ecosystem and Sleek with the last v1 branch release (v1.3.1), I have just tested v2.0.1 and I am very satisfied with how it works so far. The only issue I got was that dark mode initially was not working properly (it was still light mode even after restarting the app), but after toggling the theme option in the Settings dialog and restarting several times, it ended up working fine.

For the rest, it is an overall improvement so far. The v1 branch could not add creation dates, v2 now can, which makes it fully compatible with lots of other todo.txt managers on other platforms such as SimpleTask on Android. The interface is very responsive and nice (quite similar to before but slightly more streamlined). I like that the tag filter is remembered when I restart the app. Also I like that I can customize the reordering of filters such as priority etc. Similarly as what SimpleTask offers.

I would just like to suggest to add the possibility to save a combination of filters as a shortcut (inside the app, it could be another subtab inside the filters tab), so that we could very quickly switch between different views of the same task list (eg, one shortcut to reorder by due dates, another to order by priority, another by tasks with the @next tag, another by inversely ordered creation date to see very old tasks that may need to be trimmed, etc). But the app is very usable as it is IMHO, it's not necessary but it would be an awesome plus for power users.

Thank you so very much @ransome1 and contributors, you are making an outstanding job with this open-source tool!

/EDIT: I ran Sleek on 2 Windows 10 and Windows 11 machines.

**Comment by ransome1** [comment]
Thanks for your valuable feedback. I'm surprised the theme bug still happens. I had hoped to have that fixed, but it still seems to persists. I'll look into it.

As for your filter suggestion, I believe a similar feature request had been added to the backlog some time ago. Let's see what we can do about it.

**Comment by mdbdb** [comment]
I have been using sleek and the todo.txt for a few years now. I was quite happy that I could already install the stable version via flatpak under Linux this morning.

The first attempts were quite good, I am quite satisfied so far. Only one thing I could not solve yet:

After adjusting some settings according to my needs, I noticed that sleek no longer displays all tasks: My todo.txt file contains several entries whose due dates extend far into the next year. After I found the setting to also show future entries, I still only see 42 entries. In the search field, however, I am shown that there are 98 visible tasks. 

Did I possibly miss a setting to show **all** tasks?

All in all, very good work, thank you!

**Reply by ransome1** [reply]
Could you check in the filter settings, if "Due date in the future" is checked?

**Reply by ransome1** [reply]
@mdbdb and a another quick question, since in another thread, some users report the Flatpak release. Are you using Wayland? You can find out by typing this in your terminal: `echo $XDG_SESSION_TYPE`

**Reply by mdbdb** [reply]
- Option *Due date in the future* was and is checked
-  `echo $XDG_SESSION_TYPE` --> `x11`

But: Problem solved! I have just started my PC, started sleek and: All tasks are displayed. I made no changes to the todo.txt file.

Thanks for the quick reply anyway!

**Comment by julius-d** [comment]
Hi!

I am completely new to sleek.
I installed sleek v2.0.1 on my Mac today and I really like it. Thanks for this great tool!
As I never used any other version of sleek I can not tell if my findings are related to the v2 branch.

1. I would love to be able to add a clickable url to a todo entry. I tried it with a plain url and with markdown format but both did not work.
2. I would like to reorder todos that have the same prio. 
   https://github.com/ransome1/sleek/wiki/Keyboard-shortcuts mentions   "Move up & down todo list (when list is focused)". But I was 
   not able to find out how to focus anything.

**Reply by ransome1** [reply]
> I would love to be able to add a clickable url to a todo entry. I tried it with a plain url and with markdown format but both did not work.

The implementation of `marked` (the library that parses markdown) is still in the backlog and will be released in one of the very next releases.

> I would like to reorder todos that have the same prio.
https://github.com/ransome1/sleek/wiki/Keyboard-shortcuts mentions "Move up & down todo list (when list is focused)". But I was
not able to find out how to focus anything.

The keyboard shortcut description might be a bit misleading. It's really just about using the `tabulator` in order to set the focus on the single todo and then being able to move the focus up and down. It does not mean, you can use `arrow up` and `down` in order to apply the attributes of the next group to it. Maybe this would be an interesting feature. Feel free to provide a properly explained, detailed feature request, if you're interested in something like this. Or even better; If you are able to code, you're invited to become a contributor :)

**Reply by julius-d** [reply]
Thank you :)

**Comment by TylerRajotte** [comment]
Overall the 2.0 rewrite has been pretty good for me overall. Haven't run into any major issues. However I did notice that the option to replace links with shorter version into just the page titles has been lost in the upgrade from 1.0 -> 2.0. Maybe this feature is being looked at for a bit of an overhaul since the original 1.0 implementation used to flash the full links before they were shortened every once in a while? Just wanted to bring this up in this feedback thread as I haven't seen it written down anywhere else. Thanks

**Reply by ransome1** [reply]
Thanks for your feedback. The missing markdown feature is already in the backlog of 2.0.3. This will take some time now since I'm afk for a bid, but should be available before December.

https://github.com/ransome1/sleek/issues/563

**Comment by djihz** [comment]
Hi,
I discovered and installed Sleek yesterday (2.0.2 on M2 macbook air/macOS 14.0), after a rather long and strained search for a reasonable/lightweight/somewhat elegant task management system/software/service.
I believe Sleek will do the job nicely, thank you very much.

A bit of feedback then, hoping you will find it useful.

Very minor issues/observations so far:
- Drawer/Attributes list: my understanding of the normal behavior is that (1) a click on an attribute in the drawer displays only the tasks with that attribute, and (2) a cmd-click hides all tasks with said attribute.
In my case however Sleek may behave like (2) without pressing cmd or any modifier key.
When that happens I can return Sleek to its normal state by pressing cmd once.
This seems to be triggered by using the cmd key while in Sleek to call a system command (e.g. cmd+tab to switch apps).

- Spellchecking: when creating/editing a task the text seems to be spellchecked ("wrong" words underlined with red dots). I see no option to display suggestions, accept or reject them, and no setting to disable spellchecking entirely.

- Completed tasks: today I marked my first two tasks as completed and they both disappeared completely from the list (instead of being greyed out and sorted differently like i expected). Settings/Completed todos was on. I restarted Sleek and both tasks were then displayed like expected. So far I coul not repeat this.

- EDIT: one more... when creating a task, if I add an existing context or project, Sleek will suggest matching ones. Pressing Enter will validate the suggestion, pressing Enter again afterwards will have not effect (expected behavior would be to close the dialog and create the task). I have to use the mouse to click on "Add".

Suggestions (my own tastes or opinions so feel free to disregard)
- I'd like to join the voices in thread #217 to add support for #tags. For now I use +projects attributes as tags as well, but in my view this introduces more confusion than having a third qualifier category besides contexts and projects.

- I would suggest a clearer phrasing in Drawer/Filters (e.g "Display Completed Todos" instead of "Completed Todos" etc). I found myself doubting the meaning and effect of the toggles while trying to display completed tasks that did not appear for an unrelated reason (issue above).

- Nitpicking : in a distant future I would suggest using an icon without text in it. I find that it is an unwanted attention grabber when using the task switcher for example.

Nitpicking indeed, most likely a sign that I have said enough.
Thanks again !
(Oh, and I would gladly pay for the app on the app store...)

**Reply by ransome1** [reply]
Thank you for this very detailed and useful feedback.

**Reply by djihz** [reply]
Well, I just had a major issue.
All the content of the todo.txt file seems to have been overwritten with the content of the done.txt file, except the few last lines.

I am not 100% sure of exactly what I was doing when it happened, I believe I was just marking a task as complete.
The whole list suddenly turned grey with completed tasks, which was weird because they were all archived to done.txt a few minutes before.
All the uncompleted tasks (70+) were gone (I checked the contents of the txt file itself).
The contents of todo.txt and done.txt are now exactly the same, the few last lines that are missing from todo.txt are the tasks I marked as completed last.

What happened, I have no clue.
Nothing special in my setup, except maybe that the folder where the txt files are stored is synced to the cloud (nextcloud), so there is a weak but real possibility that this is a sync issue.
Happy to open another thread if this one is not appropriate.

EDIT : there was a weird error a few minutes before when I attempted to archive completed todos. Don't remember the exact terms but it asked in red that I open "done.txt", which was already open. I closed done.txt, reopened it and the archive action worked then.

Thanks,
N

**Reply by ransome1** [reply]
That does not sound too good :/ so you have the done.txt file open in sleek? And you think it is somehow related to the archiving function?

Is it possible that you try to archive while the done.txt file was open in sleek?

**Reply by djihz** [reply]
Actually after installing sleek, the first time I tried to archive I got an error saying that the file done.txt was not open. 
I then created done.txt in the same directory as todo.txt and tried again, got the same error. 
So I opened done.txt in Sleek and it worked fine after that. 

Being unfamiliar with Sleek and more generally the todo.txt method I thought having done.txt open was possibly a weird requirement, I understand now it is not.
So yes, done.txt stayed permanently open in Sleek since I first started using it a few days ago, including the times when I used the archive command.

I got the same error this morning a few minutes before the crash, only this time done.txt existed and was already open.
This wasn't normal but I didn't think much of it, restarted Sleek and archived without issue.

Now I just tried archiving with done.txt closed and it works just fine, so I am a bit confused indeed :)
I'll try keeping it closed from now on.

> you think it is somehow related to the archiving function?

Well I don't know if the crash is related but it still looks like Sleek accessing or writing to done.txt is troublesome on my system (errors when the file does not exist/is closed/is already open).

**Reply by djihz** [reply]
A bit more information :

Today sleek stopped showing tasks without priority in the list (but they were included in the "visible todos" count in the search bar).
I restarted sleek, and everything went back to normal.

I then archived the completed todos, and the command failed with the following error: `EPERM: operation not permitted, open '/Users/nico/Sleek/done.txt'`, which is the same error I mentioned in my feedback above, only now I was ready to take a screenshot and get the full text.
Trouble with macOS permissions ?
Hope this helps,
N

**Reply by ransome1** [reply]
Did you install it from the Mac App Store? Then yes, it is a permission issue. sleek cannot just archive into any file, that you yourself haven't selected with the file chooser. That's part of Apples sandbox system. Until I found a proper design solution for this, a workaround will most likely be to right click on the file tab of your todo file and reselect the done file using the file chooser. Let me know if this solves the permission issue.

**Reply by djihz** [reply]
Yes I did install sleek from the mac app store.
This error is the reason why I kept done.txt open, but your workaround does indeed work, thanks..

**Comment by fredofredofred** [comment]
Hi,

I used Sleek on Windows 10 before the update and am still using it after the update.

Very superficial feedback here:
- The favicon type logo is pixelated 
- I think it was possible to hide the menu bar previously, but now it isn't.
- It forgot where my .txt files were (when I opened it it was empty and I had to relocate them - really not a problem)

I hope some of that is useful. Sleek is great, thanks so much for developing this!

Fred

**Comment by qquq** [comment]
Hi, some feedback after going from v1 to v2.

Its not possible to disable context filtering/sorting anymore when clicking on "Context" headline in drawer. I used to do it to quickly focus to tasks without context. These reside now in the bottom of the list and i need to scroll all the way down to see them or, alternatively, CTRL+click on all of my context labels. Both is tedious in comparison with the previous state.
The same behaviour applies to all headlines in the drawer. They dont respond to the click like they responded in previous version, all they do is to hide or show labels.
![image](https://github.com/ransome1/sleek/assets/33117021/b2908858-3556-4513-a3e4-5a9a78d9aad4)

Still missing sorting by name, eg. alphabetically.
![image](https://github.com/ransome1/sleek/assets/33117021/8c3061e2-48b6-4c6b-86ce-c97a3ecc930c)

**Comment by fusilli** [comment]
Coming from v1 the thing I miss in v2 is the ability to toggle sorting completed tasks at the end. Other than that everything seems great! Thanks for your hard work!

**Comment by talkerswe** [comment]
I don't know of if i'm wrong here but i think there was an option to set "compact view" before, is that correct? I feel that i miss that, and if it isn't something that used to exist just ignore me! My memory is not its best!

**Comment by ransome1** [comment]
I really appreciate all the suggestions and reports. Thank you so much for it. Unfortunately this topic is side tracking the Bug report and Feature Requests process. It is hard and time intense to identify requirements from all the threads. I extracted the most urgent ones and manually created issues based off them. I will now lock the discussion here, to avoid more issue reports in this channel. I hope you don't mind.
